libkpod/image/copy.go

```diff
@@ -152,9 +158,20 @@ func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *typ
 				}
 			}
 		}
-	} else if splitArr[0] == "oci" {
-		// needs to be implemented in future
-		return errors.Errorf("oci not supported")
+	} else if splitArr[0] == "oci-archive" {
+		// generate image reference with specified image:tag
+		if len(options.OciImage) > 0 {
+			srcRef, err = ociarchive.NewReference(splitArr[len(splitArr)-1], options.OciImage)
+			if err != nil {
+				return errors.Wrapf(err, "error parsing image name %q", options.OciImage)
+			}
```

Comment:
Is the `OciImage` option truly necessary to special-case this way? AFAICS `imgName` is expected to contain an arbitrary `alltransports.ParseImageName` input, so it could just as well contain the `:`+ociImage suffix.

That would eliminate this `if`, and the non-trivial `splitArr[]` string extraction.

AFAICS this could easily happen in `loadImage`:
```go
src = ociImage + opts.Input
if opts.image != "" {
 src = src + ":" + opts.image
}
```

(or, perhaps, use a typed `types.ImageReference` for the input, and move the parsing into the caller, or split `PullImage` into two variants, one typed, one with a string input.  This all somewhat depends on how the `for _, image := range images` loop is ultimately supposed to work; maybe that would _really_ want to edit the string input, and then it’s easier to just give it strings than to bother with `containers/image/transports/ImageName` or `types.ImageReference.StringWithinTransport`)

Reply:
Agreed, fixed.